src/components/modes/student/StudentButtons.js

```diff
@@ -91,16 +124,71 @@ class StudentButtons extends Component {
     }
   };
 
+  handleSaveInput = () => {
+    const {
+      dispatchPatchAppInstanceResource,
+      dispatchPostAppInstanceResource,
+      stdinResourceId,
+      userId,
+      currentInput,
+    } = this.props;
+
+    // if there is a resource id already, update, otherwise create
+    if (stdinResourceId) {
+      dispatchPatchAppInstanceResource({
+        data: currentInput,
+        id: stdinResourceId,
+      });
+    } else {
+      dispatchPostAppInstanceResource({
+        data: currentInput,
+        type: STDIN,
+        userId,
+      });
+    }
+  };
+
   handleRun = () => {
-    const { currentCode, dispatchRunCode } = this.props;
-    dispatchRunCode(currentCode);
+    const { currentInput, currentCode, dispatchRunCode } = this.props;
+    const job = {
+      data: currentCode,
+      input: currentInput,
+    };
+
+    dispatchRunCode(job);
+  };
+
+  handleOpenInput = () => {
```

Comment:
If this works both for opening and closing, it should be called `handleToggleInput`.

Reply:
DONE.